crates/loans/src/lend_token.rs

```diff
@@ -135,16 +135,12 @@ impl<T: Config> Pallet<T> {
 
         // Formula
         // reducible_underlying_amount = liquidity / collateral_factor / price
-        let price = Self::get_price(underlying_id)?;
-
         let reducible_supply_value = liquidity
             .checked_div(&market.collateral_factor.into())
             .ok_or(ArithmeticError::Overflow)?;
-
-        let reducible_underlying_amount = reducible_supply_value
-            .checked_div(&price)
-            .ok_or(ArithmeticError::Underflow)?
-            .into_inner();
+        let reducible_supply_amount =
+            Amount::<T>::from_unsigned_fixed_point(reducible_supply_value, T::ReferenceAssetId::get())?;
```

Comment:
Would it be possible to have `get_account_liquidity`  return an `Amount`?

Reply:
I just tried it now but it has a lot of ramifications. Switching to your suggestion fixes https://github.com/interlay/interbtc/issues/751 and I'll do this after the testnet launch.